controllers/vmware/vspherecluster_reconciler.go

```diff
@@ -369,9 +370,68 @@ func (r *ClusterReconciler) VSphereMachineToCluster(ctx context.Context, o clien
 	}}
 }
 
+// ZoneToVSphereClusters adds reconcile requests for VSphereClusters when Zone has an event.
+func (r *ClusterReconciler) ZoneToVSphereClusters(ctx context.Context, o client.Object) []reconcile.Request {
+	log := ctrl.LoggerFrom(ctx)
+
+	zone, ok := o.(*topologyv1.Zone)
+	if !ok {
+		log.Error(nil, fmt.Sprintf("Expected a Zone but got a %T", o))
+		return nil
+	}
+	log = log.WithValues("Zone", klog.KObj(zone))
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	VSphereClusters := &vmwarev1.VSphereClusterList{}
```

Comment:
```suggestion
	vSphereClusters := &vmwarev1.VSphereClusterList{}
```

Reply:
How about `vsphereCluster`, I noticed our existing code is using this format.

Reply:
That's fine!

Reply:
Based on https://www.google.de/search?q=vSphere&sca_esv=6cfb153aab81f25c&sca_upv=1&source=hp&ei=kd66ZprqIM_Oxc8P2uvmuQc&iflsig=AL9hbdgAAAAAZrrsoTQdNNqqr1hp3-lM4RlIunN08bnx&ved=0ahUKEwja9oC1jvGHAxVPZ_EDHdq1OXcQ4dUDCBg&uact=5&oq=vSphere&gs_lp=Egdnd3Mtd2l6Igd2U3BoZXJlMgUQABiABDIFEAAYgAQyBRAAGIAEMgUQABiABDIFEAAYgAQyBRAAGIAEMgUQABiABDIFEAAYgAQyBRAAGIAEMgUQABiABEjkDFAAWOgGcAB4AJABAJgBfKABqQaqAQMwLje4AQPIAQD4AQGYAgegAsAGwgILEC4YgAQY0QMYxwHCAgUQLhiABMICCxAuGIAEGMcBGK8BmAMAkgcDMC43oAe3LQ&sclient=gws-wiz

vSphere seems to be correct, but let's just stay consistent for this PR and not do another iteration :)